### PR TITLE
refactor(versioning/docker): Enable strict null checks

### PR DESCRIPTION
**Pull request by zharinov**
## Changes:

- Enable strict null checks

## Context:

- Ref: #7154

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — DISMISSED** [review]

File: lib/datasource/docker/__snapshots__/index.spec.ts.snap [hunk 1]
```diff
@@ -22,10 +22,10 @@ Object {
       "version": "1.0.0",
     },
     Object {
-      "version": "1.2.3-alpine",
+      "version": "1.2.3",
     },
     Object {
-      "version": "1.2.3",
+      "version": "1.2.3-alpine",
```
Comment (zharinov) [inline lib/datasource/docker/__snapshots__/index.spec.ts.snap:28]: Should be safe change

File: lib/versioning/docker/index.ts [hunk 2]
```diff
@@ -65,16 +64,24 @@ class DockerVersioningApi extends GenericVersioningApi {
...
+        return parsed1.prerelease.localeCompare(parsed2.prerelease);
+      }
     }
     // equals
-    return parsed2.suffix.localeCompare(parsed1.suffix);
+    if (parsed2.suffix && parsed1.suffix) {
+      return parsed2.suffix.localeCompare(parsed1.suffix);
+    }
+
+    return 0;
```
Comment (zharinov) [inline lib/versioning/docker/index.ts:76]: Snapshot change was caused by this line

File: lib/datasource/docker/__snapshots__/index.spec.ts.snap (hunk 1 above)
Comment (zharinov) [reply lib/datasource/docker/__snapshots__/index.spec.ts.snap:28]: No, it isn't

File: lib/versioning/docker/index.ts (hunk 2 above)
Comment (viceice) [reply lib/versioning/docker/index.ts:76]: This is wrong, you need to check for empty string in suffix, so your change makes `1.2.3 === 1.2.3-alpine`. Same for prerelease

**Review by rarkins — APPROVED** [review]

File: lib/versioning/docker/index.ts [hunk 3]
```diff
@@ -65,16 +64,23 @@ class DockerVersioningApi extends GenericVersioningApi {
         return -1;
       }
       // alphabetic order
-      return parsed1.prerelease.localeCompare(parsed2.prerelease);
+      if (parsed1.prerelease && parsed2.prerelease) {
+        return parsed1.prerelease.localeCompare(parsed2.prerelease);
```
Comment (viceice) [inline lib/versioning/docker/index.ts:68]: Do we need to check for empty string here too?

**Review by viceice — COMMENTED** [review]
Otherwise LGTM

**Pull request merged**